### PR TITLE
Update daily leaderboard in the app #178

### PR DESCRIPTION
**Pull request by satyamurti**

### CODE REVIEW

File: app/src/main/java/org/pragyan/dalal18/fragment/LeaderboardFragment.kt [hunk 1]
```diff
@@ -105,6 +123,7 @@ class LeaderboardFragment : Fragment() {
                 }
 
                 for (rankListResponse in rankListResponses) {
+                    d("SPD", rankListResponse.statusCode.toString())
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/LeaderboardFragment.kt:126]: Remove logs

File: app/src/main/java/org/pragyan/dalal18/fragment/LeaderboardFragment.kt [hunk 2]
```diff
@@ -73,21 +80,32 @@ class LeaderboardFragment : Fragment() {
...
-            adapter = leaderBoardRecyclerAdapter
-        }
+//        getRankListAsynchronously()
+//        leaderBoardRecyclerAdapter = LeaderboardRecyclerAdapter(context, leaderBoardDetailsList)
+//
+//        with(binding.leaderboardRecyclerView) {
+//            layoutManager = LinearLayoutManager(context)
+//            setHasFixedSize(false)
+//            adapter = leaderBoardRecyclerAdapter
+//        }
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/LeaderboardFragment.kt:108]: Remove comments if not in use

File: app/src/main/java/org/pragyan/dalal18/fragment/LeaderboardFragment.kt [hunk 3]
```diff
@@ -137,4 +195,31 @@ class LeaderboardFragment : Fragment() {
     companion object {
         private const val LEADER_BOARD_SIZE = 100
     }
+
+    override fun onNothingSelected(p0: AdapterView<*>?) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onItemSelected(p0:     AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/LeaderboardFragment.kt:203]:
```suggestion
    override fun onItemSelected(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
```

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -478,13 +476,17 @@
     <string name="cash">Cash</string>
     <string name="reserved_cash">Reserved Cash</string>
     <string name="trade_price">Trade Price</string>
-    <string name="referral_code">Have a referral Code? Enter it here.</string>
```
Comment (avikumar15) [inline app/src/main/res/values/strings.xml:481]: Don't delete these.

File: app/src/main/java/org/pragyan/dalal18/fragment/ReferAndEarnFragment.kt [hunk 5]
```diff
@@ -41,7 +40,7 @@ class ReferAndEarnFragment : Fragment() {
 
     private var referralCodeDialog: AlertDialog? = null
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/fragment/ReferAndEarnFragment.kt:43]: Revert this?

File: app/src/main/java/org/pragyan/dalal18/fragment/CompanyDescriptionFragment.kt [hunk 6]
```diff
@@ -17,7 +17,7 @@ class CompanyDescriptionFragment : Fragment() {
 
     private lateinit var model: DalalViewModel
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/fragment/CompanyDescriptionFragment.kt:20]: any specific reason to make the view **non-nullable** here ?
refer to the official [view-binding](https://developer.android.com/topic/libraries/view-binding) docs please !

File: app/src/main/java/org/pragyan/dalal18/utils/Constants.java [hunk 7]
```diff
@@ -5,7 +5,7 @@
     // TODO (Release): Change server IP and Port
 
     public static final String HOST = "dalal.pragyan.org";
-    public static final int PORT = 8080;
+    public static final int PORT = 8000;
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/utils/Constants.java:8]: revert this to original configuration.

**Review by ranaaditya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/pragyan/dalal18/utils/Constants.java (hunk 7 above)
Comment (Sudhindra3000) [reply app/src/main/java/org/pragyan/dalal18/utils/Constants.java:8]: But the server is running at Port 8000. Original port number has changed in Server

File: app/src/main/java/org/pragyan/dalal18/utils/Constants.java [hunk 8]
```diff
@@ -4,7 +4,7 @@
 
     // TODO (Release): Change server IP and Port
 
-    public static final String HOST = "dalal.pragyan.org";
+    public static final String HOST = "192.168.0.110";
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/utils/Constants.java:7]: revert this to **dalal.pragyan.org**

**Review by ranaaditya — CHANGES_REQUESTED** [review]

**Review by ranaaditya — APPROVED** [review]

**Review by ranaaditya — APPROVED** [review]

**Pull request merged**